### PR TITLE
Add per channel untethered window limit and resting timeouts

### PR DESCRIPTION
**Pull request by mikeb01**
* Additional refactoring/cleanup of C++ ChannelUriStringBuilder.

### CODE REVIEW

File: aeron-client/src/main/c/uri/aeron_uri.h [hunk 1]
```diff
@@ -73,6 +73,8 @@ aeron_uri_params_t;
 #define AERON_URI_TIMESTAMP_OFFSET_RESERVED "reserved"
 #define AERON_URI_RESPONSE_CORRELATION_ID_KEY "response-correlation-id"
 #define AERON_URI_NAK_DELAY_KEY "nak-delay"
+#define AERON_URI_UNTETHERED_WINDOW_LIMIT_TIMEOUT_KEY "untethered-window-limit-timeout"
+#define AERON_URI_UNTETHERED_RESTING_TIMEOUT_KEY "untethered-window-limit-timeout"
```
Comment (vyazelenko) [inline aeron-client/src/main/c/uri/aeron_uri.h:77]: Copy paste error, i.e. should be `untethered-resting-timeout` instead.

File: aeron-driver/src/main/java/io/aeron/driver/PublicationParams.java [hunk 2]
```diff
@@ -386,6 +390,22 @@ private void getSpiesSimulateConnection(final ChannelUri channelUri, final Media
...
+    }
+
+    private void getUntetheredRestingTimeout(final ChannelUri channelUri, final MediaDriver.Context ctx)
+    {
+        final String timeoutString = channelUri.get(UNTETHERED_RESTING_TIMEOUT_PARAM_NAME);
+        untetheredWindowLimitTimeoutNs = null != timeoutString ?
+            SystemUtil.parseDuration(UNTETHERED_RESTING_TIMEOUT_PARAM_NAME, timeoutString) :
+            ctx.untetheredWindowLimitTimeoutNs();
+    }
+
```
Comment (vyazelenko) [inline aeron-driver/src/main/java/io/aeron/driver/PublicationParams.java:408]: Could be refactored to a single method similar to `PublicationImage.getTimeoutNsFromChannel`. Maybe we could add a method to `ChannelUri`?

File: aeron-driver/src/main/java/io/aeron/driver/PublicationParams.java [hunk 3]
```diff
@@ -386,6 +390,22 @@ private void getSpiesSimulateConnection(final ChannelUri channelUri, final Media
...
+            SystemUtil.parseDuration(UNTETHERED_WINDOW_LIMIT_TIMEOUT_PARAM_NAME, timeoutString) :
+            ctx.untetheredWindowLimitTimeoutNs();
+    }
+
+    private void getUntetheredRestingTimeout(final ChannelUri channelUri, final MediaDriver.Context ctx)
+    {
+        final String timeoutString = channelUri.get(UNTETHERED_RESTING_TIMEOUT_PARAM_NAME);
+        untetheredWindowLimitTimeoutNs = null != timeoutString ?
+            SystemUtil.parseDuration(UNTETHERED_RESTING_TIMEOUT_PARAM_NAME, timeoutString) :
+            ctx.untetheredWindowLimitTimeoutNs();
```
Comment (vyazelenko) [inline aeron-driver/src/main/java/io/aeron/driver/PublicationParams.java:406]: The default value should be `ctx.untetheredRestingTimeoutNs()`.

**Review by vyazelenko — CHANGES_REQUESTED** [review]

**Comment by vyazelenko** [comment]
@mikeb01 The MacOS build is fixed on master.

File: aeron-driver/src/main/java/io/aeron/driver/PublicationParams.java (hunk 2 above)
Comment (mikeb01) [reply aeron-driver/src/main/java/io/aeron/driver/PublicationParams.java:408]: I've done the refactoring, but I'm going to leave it in the current place for now.

File: aeron-client/src/main/cpp_wrapper/ChannelUri.h [hunk 4]
```diff
@@ -68,6 +68,8 @@ static constexpr const char MEDIA_RCV_TIMESTAMP_OFFSET_PARAM_NAME[] = "media-rcv
 static constexpr const char CHANNEL_RCV_TIMESTAMP_OFFSET_PARAM_NAME[] = "channel-rcv-ts-offset";
 static constexpr const char CHANNEL_SND_TIMESTAMP_OFFSET_PARAM_NAME[] = "channel-snd-ts-offset";
 static constexpr const char RESPONSE_CORRELATION_ID_PARAM_NAME[] = "response-correlation-id";
+static constexpr const char UNTETHERED_WINDOW_LIMIT_TIMEOUT_PARAM_NAME[] = "untethered-window-limit-timeout";
+static constexpr const char UNTETHERED_RESTING_TIMEOUT_PARAM_NAME[] = "untethered-window-limit-timeout";
```
Comment (vyazelenko) [inline aeron-client/src/main/cpp_wrapper/ChannelUri.h:72]: Constant name and the value do not match.
Should be:
```cpp
static constexpr const char UNTETHERED_RESTING_TIMEOUT_PARAM_NAME[] = "untethered-resting-limit-timeout";
```

**Review by vyazelenko — CHANGES_REQUESTED** [review]

File: aeron-client/src/main/cpp_wrapper/ChannelUri.h (hunk 4 above)
Comment (mikeb01) [reply aeron-client/src/main/cpp_wrapper/ChannelUri.h]: I had missed pushing a couple of commits that picked that up.

**Pull request merged**